src/Roslyn.Diagnostics.Analyzers/CSharp/CSharpAvoidOptSuffixForNullableEnableCode.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.Lightup;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Diagnostics.Analyzers;
+
+namespace Roslyn.Diagnostics.CSharp.Analyzers
+{
+    /// <summary>
+    /// RS0046: Avoid 'Opt' suffix for nullable enable code
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpAvoidOptSuffixForNullableEnableCode : DiagnosticAnalyzer
```

Comment:
Why not inherit from a base "AvoidOptSuffixForNullableEnableCode" (that will inturn inherit from DiagnosticAnalyzer) instead of direct inheritance from DiagnosticAnalyzer?

That will allow easily adding a VB analyzer.

Reply:
I started with the base class but after some chat we decided it was unlikely that the case will happen for vbnet. Note that this analyzer is mainly (only) meant to help on roslyn.